### PR TITLE
IDENTITY-3500 block addRole when a assigned user is in delete workflow

### PR DESCRIPTION
**Pull request by cdwijayarathna**

### CODE REVIEW

File: components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/AddRoleWFRequestHandler.java [hunk 1]
```diff
@@ -220,6 +220,10 @@ public boolean isValidOperation(Entity[] entities) throws WorkflowException {
                         workflowService.entityHasPendingWorkflowsOfType(entities[i], UserStoreWFConstants
                                 .UPDATE_ROLE_NAME_EVENT))) {
                     throw new WorkflowException("Rolename already exists in the system. Please pick another rolename.");
+                } else if (workflowService.eventEngagedWithWorkflows(UserStoreWFConstants.ADD_USER_EVENT) &&
+                        entities[i].getEntityType() == UserStoreWFConstants.ENTITY_TYPE_USER && workflowService
```
Comment (lalaji) [inline components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/AddRoleWFRequestHandler.java]: Isn't that we have to use 'equals()' instead '==' as we are doing a object content comparison?

File: components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/UpdateRoleNameWFRequestHandler.java [hunk 2]
```diff
@@ -178,14 +176,10 @@ public boolean isValidOperation(Entity[] entities) throws WorkflowException {
 
         WorkflowService workflowService = IdentityWorkflowDataHolder.getInstance().getWorkflowService();
         for (int i = 0; i < entities.length; i++) {
-            try {
-                if (entities[i].getEntityType() == UserStoreWFConstants.ENTITY_TYPE_ROLE && workflowService
+            if (entities[i].getEntityType() == UserStoreWFConstants.ENTITY_TYPE_ROLE && workflowService
```
Comment (lalaji) [inline components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/UpdateRoleNameWFRequestHandler.java]: Isn't that we have to use 'equals()' instead '==' as we are doing a object content comparison?

File: components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/AddRoleWFRequestHandler.java (hunk 1 above)
Comment (cdwijayarathna) [inline components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/AddRoleWFRequestHandler.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1699

File: components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/UpdateRoleNameWFRequestHandler.java (hunk 2 above)
Comment (cdwijayarathna) [inline components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/UpdateRoleNameWFRequestHandler.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1699

**Pull request merged**